inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -196,7 +196,135 @@ std::shared_ptr<ICNNNetwork> V10Parser::parse(const pugi::xml_node& root, std::i
             result_nodes[0]->add_control_dependency(assign);
         }
     }
-    return CNNNetwork(function);
+    CNNNetwork net(function);
+    parsePreProcess(net, root, binStream);
+    return net;
+}
+
+void V10Parser::parsePreProcess(CNNNetwork& network, const pugi::xml_node& root, std::istream& binStream) {
+    /*
+        <pre-process mean-precision="FP32">
+        <channel id = ”0”>
+        <mean offset = "121930449" size = "51529" / >  // in case of array – ref to the .bin file
+        </channel>
+        </pre-process>
+    */
+
+    auto ppNode = root.child("pre-process");
+    if (ppNode.empty()) {
+        return;
+    }
+    // find out to what input this belongs to
+    std::string inputName;
+    InputInfo::Ptr preProcessInput;
+
+    inputName = GetStrAttr(ppNode, "reference-layer-name", "");
+    inputName = ngraph::trim(inputName);
+    if (inputName.empty()) {
+        // fallback (old format), look for the picture in the inputs
+        InputsDataMap inputs = network.getInputsInfo();
+
+        if (inputs.empty()) THROW_IE_EXCEPTION << "network has no input";
+
+        for (auto i : inputs) {
+            if (i.second->getTensorDesc().getDims().size() == 4) {
+                preProcessInput = i.second;
+                break;
+            }
+        }
+        if (!preProcessInput) {
+            preProcessInput = inputs.begin()->second;
+        }
+
+        inputName = preProcessInput->name();
+    } else {
+        preProcessInput = network.getInputsInfo().at(inputName);
+        if (!preProcessInput)
```

Comment:
std::map will throw an out_of_range exception if key doesn't exists. So this statement won't work.

Reply:
Agree, fixed